src/store/hooks/useAuth.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  createContext, useContext, useMemo, ReactNode,
+  useEffect, useState,
+} from 'react';
+import {
+  getAuth, onAuthStateChanged, User, signOut, Auth,
+} from 'firebase/auth';
+import { LoadingOverlay } from '@mantine/core';
+import { useStorageEngine } from '../storageEngineHooks';
+import { FirebaseStorageEngine } from '../../storage/engines/FirebaseStorageEngine';
+import { GlobalConfig } from '../../parser/types';
+
+interface LocalStorageUser {
+  name: string,
+  email: string
+}
+
+type UserOptions = User | LocalStorageUser | null;
+
+interface UserWrapped {
+  user: UserOptions,
+  determiningStatus: boolean,
+  isAdmin: boolean,
+  adminVerification:boolean
+}
+
+// Defines default AuthContextValue
+interface AuthContextValue {
+  user: UserWrapped;
+  logout: () => void;
+  verifyAdminStatus: (firebaseUser: UserOptions) => Promise<boolean>;
+  }
+
+// Initializes AuthContext
+const AuthContext = createContext<AuthContextValue>({
+  user: {
+    user: null,
+    determiningStatus: false,
+    isAdmin: false,
+    adminVerification: false,
+  },
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  logout: () => {},
+  verifyAdminStatus: () => Promise.resolve(false),
+
+});
+
+// Firebase auth context
+export const useAuth = () => useContext(AuthContext);
+
+// Defines the functions that are exposed in this hook.
+export function AuthProvider({ children, globalConfig } : { children: ReactNode, globalConfig: GlobalConfig }) {
+  const [user, setUser] = useState<UserWrapped>(
+    {
+      user: null,
+      determiningStatus: true,
+      isAdmin: false,
+      adminVerification: false,
+    },
+  );
+
+  // Default non-user when loading
+  const loadingNullUser : UserWrapped = {
+    user: null,
+    determiningStatus: true,
+    isAdmin: false,
+    adminVerification: false,
+  };
+
+  // Default non-user when not loading
+  const nonLoadingNullUser : UserWrapped = {
+    user: null,
+    determiningStatus: false,
+    isAdmin: false,
+    adminVerification: false,
+  };
+
+  // Non-auth User
+  const nonAuthUser : UserWrapped = {
+    user: {
+      name: 'fakeName',
+      email: 'fakeEmail@fake.com',
+    },
+    determiningStatus: false,
+    isAdmin: true,
+    adminVerification: true,
+  };
```

Comment:
A way to use the user objects that you create

```suggestion
  // Default non-user when loading
  const loadingNullUser : UserWrapped = {
    user: null,
    determiningStatus: true,
    isAdmin: false,
    adminVerification: false,
  };

  // Default non-user when not loading
  const nonLoadingNullUser : UserWrapped = {
    user: null,
    determiningStatus: false,
    isAdmin: false,
    adminVerification: false,
  };

  // Non-auth User
  const nonAuthUser : UserWrapped = {
    user: {
      name: 'fakeName',
      email: 'fakeEmail@fake.com',
    },
    determiningStatus: false,
    isAdmin: true,
    adminVerification: true,
  };
  
  const [user, setUser] = useState(loadingNullUser);
```